app/src/main/native/cpp/looper_thread.cpp

```diff
@@ -49,42 +56,26 @@ int LooperThread::looperCallback(int fd, int events, void * data) {
     LOGW("Handling more than one event is not implemented and "
          "should not happen in current implementation. Only one event will be processed.");
   }
-  char buffer[1];
-  while (read(fd, buffer, sizeof(buffer)) > 0) {}
-  auto taskId = *reinterpret_cast<uint8_t*>(buffer);
+  char buffer[sizeof(uintptr_t)];
+  while (read(fd, buffer, sizeof(uintptr_t)) > 0) {}
+  auto taskId = *reinterpret_cast<uintptr_t*>(buffer);
   auto * looperThread = reinterpret_cast<LooperThread*>(data);
-  // treating internal destroy task a bit differently
-  if (taskId == STOP_THREAD_TASK_ID) {
-    ALooper_removeFd(looperThread->aLooper, fd);
-    if (close(looperThread->fds[PIPE_IN]) || close(looperThread->fds[PIPE_OUT])) {
-      throw std::runtime_error("Failed to close file descriptor!");
-    }
-    // explicit wake here in order to unblock ALooper_pollAll
-    ALooper_wake(looperThread->aLooper);
-    ALooper_release(looperThread->aLooper);
-    // returning 0 to have this file descriptor and callback unregistered from the looper
-    return 0;
-  }
-  if (looperThread->methodMap.find(taskId) != looperThread->methodMap.end()) {
-    looperThread->methodMap[taskId]();
-  } else {
-    LOGW("Looper callback triggered with taskId=%i, but associated function could not be found!", taskId);
+  static tbb::concurrent_hash_map<uintptr_t, std::function<void()>>::accessor ac;
```

Comment:
nit: you could probably create an alias for this type: `using Accessor = tbb::concurrent_hash_map<uintptr_t, std::function<void()>>::accessor` and use it instead.